x-pack/plugins/lists/common/schemas/types/endpoint/entries.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable @typescript-eslint/camelcase */
+
+import * as t from 'io-ts';
+import { Either } from 'fp-ts/lib/Either';
+
+import { endpointEntryMatchAny } from './entry_match_any';
+import { endpointEntryMatch } from './entry_match';
+import { endpointEntryNested } from './entry_nested';
+
+export const endpointEntriesArray = t.array(
+  t.union([endpointEntryMatch, endpointEntryMatchAny, endpointEntryNested])
+);
+export type EndpointEntriesArray = t.TypeOf<typeof endpointEntriesArray>;
+
+/**
+ * Types the nonEmptyEndpointEntriesArray as:
+ *   - An array of entries of length 1 or greater
+ *
+ */
+export const nonEmptyEndpointEntriesArray = new t.Type<
+  EndpointEntriesArray,
+  EndpointEntriesArray,
+  unknown
+>(
+  'NonEmptyEndpointEntriesArray',
+  endpointEntriesArray.is,
```

Comment:
I think this type guard needs to be more specific... the type guard could pass for an array of length 0, which would fail to decode to `endpointEntriesArray`.

Reply:
Good point, i think we have the same problem with the generic exceptions [here](https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/common/schemas/types/non_empty_nested_entries_array.ts#L28)
I'll fix the endpoint type guards in this PR and look at the generic ones separately.